### PR TITLE
add keras doc

### PR DESCRIPTION
**Pull request by xiaxin1998**
./MaxPooling1D.scala
--
./SparseEmbedding.scala
./SpatialDropout3D.scala
./Masking.scala
./AveragePooling3D.scala
./GlobalMaxPooling1D.scala
./PReLU.scala
./LocallyConnected2D.scala
./Recurrent.scala
./MaxPooling2D.scala

### CODE REVIEW

**Comment by xiaxin1998** [comment]
@songhappy  Please review my pr. Thanks.

**Comment by xiaxin1998** [comment]
http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1486/ Jenkins passed.

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 1]
```diff
@@ -1,3 +1,70 @@
...
+Masking(maskValue = 0.0, inputShape = null)
+```
+**Python:**
+```python
+Masking(mask_value=0.0, input_shape=None, name=None)
+```
+
+**Parameters:**
+
+* `maskValue`: Mask value. For each timestep in the input (the second dimension), if all the values in the input at that timestep are equal to 'maskValue', then the timestep will masked (skipped) in all downstream layers.
```
Comment (songhappy) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: then the timestep will be masked (skipped) in all downstream layers.

File: docs/docs/KerasStyleAPIGuide/Layers/dropout.md [hunk 2]
```diff
@@ -1,3 +1,185 @@
+## **SpatialDropout3D**
+Spatial 3D version of Dropout.
+
+This version performs the same function as Dropout, however it drops entire 3D feature maps instead of individual elements. If adjacent voxels within feature maps are strongly correlated (as is normally the case in early convolution layers) then regular dropout will not regularize the activations and will otherwise just result in an effective learning rate decrease. In this case, SpatialDropout3D will help promote independence between feature maps and should be used instead.
```
Comment (songhappy) [inline docs/docs/KerasStyleAPIGuide/Layers/dropout.md]: the same functionalities

File: docs/docs/KerasStyleAPIGuide/Layers/dropout.md [hunk 3]
```diff
@@ -1,3 +1,185 @@
...
+
+The input of this layer should be 5D.
+
```
Comment (songhappy) [inline docs/docs/KerasStyleAPIGuide/Layers/dropout.md]: again, please specify each dimension

File: docs/docs/KerasStyleAPIGuide/Layers/pooling.md [hunk 4]
```diff
@@ -1,3 +1,464 @@
+## **MaxPooling1D**
+Max pooling operation for temporal data.
+
+The input of this layer should be 3D.
+
```
Comment (songhappy) [inline docs/docs/KerasStyleAPIGuide/Layers/pooling.md]: again, please specify each dimension of the input

File: docs/docs/KerasStyleAPIGuide/Layers/pooling.md [hunk 5]
```diff
@@ -1,3 +1,464 @@
...
+ [[0.9950283  0.99869156 0.877897   0.6784016  0.98447174]]]
+```
+
+---
+## **MaxPooling2D**
+Max pooling operation for spatial data.
+
+The input of this layer should be 4D.
+
```
Comment (songhappy) [inline docs/docs/KerasStyleAPIGuide/Layers/pooling.md]: specify each dimension

**Comment by xiaxin1998** [comment]
18888/view/ZOO-PR/job/ZOO-PR-Validation/1772/ Jenkins passed.

**Comment by xiaxin1998** [comment]
18888/job/ZOO-PR-Validation/1776/ Jenkins passed. @songhappy @hkvision

**Comment by hkvision** [comment]
@songhappy Please have another review.

File: docs/docs/KerasStyleAPIGuide/Layers/convolutional.md [hunk 6]
```diff
@@ -1,3 +1,137 @@
+## **LocallyConnected2D**
+A Locally-connected layers for 2D input works similarly to a SpatialConvolution layer, except that weights are unshared, that is, a different set of filters is applied at different patch of the input.
+
+The input of this layer should be 4D. The first dimension is samples, the second dimension is chanels, the third dimension is rows, the fourth dimension id cols.
+
```
Comment (songhappy) [inline docs/docs/KerasStyleAPIGuide/Layers/convolutional.md]: The input is 4D tensor with shape: (batch_size, channels, rows, cols)

File: docs/docs/KerasStyleAPIGuide/Layers/pooling.md [hunk 7]
```diff
@@ -1,3 +1,464 @@
...
+  [[[0.52306354]]]]]
+
+```
+
+---
+## **GlobalMaxPooling1D**
+Global max pooling operation for temporal data.
+
+The input of this layer should be 3D.
+
```
Comment (songhappy) [inline docs/docs/KerasStyleAPIGuide/Layers/pooling.md]: Input is a 3D tensor of (batch_size, steps, feature_dim)

**Comment by xiaxin1998** [comment]
I have modified the corresponding contents,please review. Thanks.

**Comment by xiaxin1998** [comment]
I have modified the corresponding contents,please review. Thanks. @songhappy

**Comment by xiaxin1998** [comment]
18888/view/ZOO-PR/job/ZOO-PR-Validation/1782 Jenkins passed after revising.

**Comment by songhappy** [comment]
LGTM
